acceptance/tests/lookup/merge_strategies.rb

```diff
@@ -0,0 +1,194 @@
+test_name 'C99903: merge strategies' do
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+  tmp_environment2 = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath2  = "#{environmentpath}/#{tmp_environment2}"
+
+  master_confdir = master.puppet('master')['confdir']
+  hiera_conf_backup = master.tmpfile(app_type)
+
+  teardown do
+    step "restore default global hiera.yaml" do
+      on(master, "mv #{hiera_conf_backup} #{master_confdir}/hiera.yaml")
```

Comment:
I admire the direction, but I'm not sure that skipping restart (at least HUP) is safe, due to caching in hiera and/or the compiler. So I'm ok to merge this PR, but something to think about before removing `with_puppet_running_on`.